swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items_matching_categories = []
+        for item in self.inventory:
+            if item.category == category:
+                items_matching_categories.append(item)
```

Comment:
Notice this pattern of 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten as a list comprehension as
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like
```python
        items_matching_categories = [item for item in self.inventory if item.category == category]
```

At first, this may seem more difficult to read, but comprehensions are very common python style, so try getting used to working with them!